src/cgame/cg_weapons.cpp

```diff
@@ -85,23 +85,31 @@ Load custom crosshairs specified by the user
 static void CG_LoadCustomCrosshairs()
 {
 	const char         *text_p, *token;
+	char         text[ 20000 ];
+	int          len;
+	fileHandle_t f;
 	weapon_t     weapon;
 
-	std::string filename = cg_crosshairFile.Get();
-	if ( filename.empty() )
+	len = trap_FS_FOpenFile( cg_crosshairFile.Get().c_str(), &f, fsMode_t::FS_READ );
+
+	if ( len < 0 )
 	{
 		return;
 	}
-	std::error_code err;
-	std::string text = FS::PakPath::ReadFile( filename, err );
-	if ( err )
+
+	if ( len == 0 || len + 1 >= (int) sizeof( text ) )
```

Comment:
Why not just malloc the length but put a max? I think MAX_TOKEN_CHARS might actually be enough?